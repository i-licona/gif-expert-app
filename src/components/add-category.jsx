import { useState } from "react";

export const AddCategory = ({ addCategory }) =>{
  // definir estado inicial
  const [inputValue, setInputValue] = useState('')
  // aÃ±adir evento change para el llenado del inputValue
  const inputChange = ({ target }) =>{
    setInputValue((s) => s = target.value);
  }
  // guardar la busqueda en el estado
  const saveChanges = (event) =>{
    // quitar recarga de por evento
    event.preventDefault();
    // validar que el input no este vacio
    if (inputValue == '' || inputValue == null || inputValue == undefined) {
      return;
    }
    // guardar busqueda en el arreglo
    addCategory(inputValue.trim()); 
    resetState();
  }
  // resetear estado inicial
  const resetState = () => {
    setInputValue((s) => s = '' );
  }
  /* template html */
  return(
    <form onSubmit={ (event) => saveChanges(event) } aria-label = 'form'>
      <input
        type='text'
        placeholder='Buscar gifs'
        value={ inputValue }
        onChange = { (event) => inputChange(event) }
      />
    </form>
  );
}