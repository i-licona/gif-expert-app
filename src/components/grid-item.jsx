
export const GifItem = ({ title, url }) =>{
  return(
    <div className="card">
      <img src={ url } alt="git resultado" />
    </div>
  )
}