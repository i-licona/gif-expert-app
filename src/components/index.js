export * from './add-category';
export * from './gif-grid';
export * from './grid-item';
