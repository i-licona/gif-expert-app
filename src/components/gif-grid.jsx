import React, { useEffect, useState } from "react"
import { getGifs } from "../helppers/gifs.services";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./grid-item";

export const GifGrid = ({ category }) => {
  const { images, isLoadding } = useFetchGifs(category);
  // html template
  return (
    <>
      <h3>{ category }</h3>
      {
        isLoadding && ( <h2>Cargando...</h2> )
      }
      
      <div className="card-grid">
        {/* mostrart gifs */}
        { images.map((img) => {
            return <GifItem key={ img.id } title={ img.title } url={ img.url } />
        })}
      </div>      
    </>
  )
}