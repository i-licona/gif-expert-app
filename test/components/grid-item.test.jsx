import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/grid-item";

describe('Pruebas en <GifItem>', () => {
  /* const */
  const title = 'Cake';
  const url = 'https://cake.com/cake.jpg';
  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={ title } url ={ url } />);
    expect(container).toMatchSnapshot();
  });
  test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={ title } url ={ url } />);
    // screen.debug();
    expect( screen.getByRole('img').src).toBe(url);
  });
});