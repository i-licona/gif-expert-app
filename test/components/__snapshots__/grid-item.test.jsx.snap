// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Cake"
      src="https://cake.com/cake.jpg"
    />
  </div>
</div>
`;
